### ISSUE TITLE
Reuse existing candle data

### ISSUE THREAD

**Issue by marijn111** [dema-trading-ai/engine#35]
For backtesting purposes, historical data (candles) are needed. These are fetched through an API and stored locally. 
However, this data must match the data needed for a backtest exactly in terms of pair, timeframe and date.

Currently, when the data does not match your backtest needs (lets say that you are backtesting 1 day longer then you have data), the engine will erase all data it has, and download everything again. 

This costs time and resources, so a check needs to be implemented that complements the existing data (if possible) so that it does matches the data needed for the backtest. 

In other words, if the stored data is the correct pair and timeframe, but it is missing a few days/weeks/month, download _only_ that period, instead of the full period

This functionality should be implemented somewhere here:
https://github.com/dema-trading-ai/engine/blob/3378ecf45093202875162cd0c8c999413888d497/data/datamodule.py#L88-L102

Roughly the following checks should be followed: is there data -> is it for the right pair -> is it for the right timeframe -> does it covers the backtesting period -> download missing candles only

**Comment by noudcorten** [comment]
I am currently working on fixing this in issue #18.

**Comment by keebrev** [comment]
@noudcorten please keep things separate. #18 was only meant for putting OHLCV data in dataframes, not fixing this issues with storing data

**Comment by marijn111** [comment]
Fixed by #43, closing.
